### ISSUE TITLE
unnecessary_null_check false positive

### ISSUE THREAD

**Issue by blandford** [bilelmoussaoui/gobject-lint#26]
I _think_ this is just checking for redundant timeout_id checks? It should check for the existence of `else` statements, as that logic is needed. I have the following diff with --fix

```patch
-  if (self->autohide_timeout_id != 0)
-    g_clear_handle_id (&self->autohide_timeout_id, g_source_remove);
-  else
-    update_autohide (self, TRUE);
+  g_clear_handle_id (&self->autohide_timeout_id, g_source_remove);
```
